File: app/src/main/java/ru/sicampus/bootcamp2025/ui/list/ListAdapter.java
package ru.sicampus.bootcamp2025.ui.list;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;
import java.util.function.Consumer;

import ru.sicampus.bootcamp2025.databinding.ItemVolunteerBinding;
import ru.sicampus.bootcamp2025.domain.entites.ItemUserEntity;

public class ListAdapter extends RecyclerView.Adapter<ListAdapter.ViewHolder> {

    @NonNull
    private final Consumer<Integer> onItemClick;

    private final List<ItemUserEntity> data = new ArrayList<>();

    public ListAdapter(@NonNull Consumer<Integer> onItemClick) {
        this.onItemClick = onItemClick;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(
                ItemVolunteerBinding.inflate(
                        LayoutInflater.from(parent.getContext()),
                        parent,
                        false
                ).getRoot()
        );

        /*return new ViewHolder(
                ItemVolunteerBinding.inflate(
                        LayoutInflater.from(parent.getContext()),
                        parent,
                        false
                )
        );*/
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bind(data.get(position));
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public void updateData(List<ItemUserEntity> newData) {
        //Log.d("ta)
        data.clear();
        data.addAll(newData);
        notifyDataSetChanged();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private final ItemVolunteerBinding binding;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            binding = ItemVolunteerBinding.bind(itemView);
        }
        /*public ViewHolder(@NonNull ItemVolunteerBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }*/

        public void bind(ItemUserEntity item) {
            binding.lastName.setText(item.getLastName());
            binding.firstName.setText(item.getFirstName());
            binding.getRoot().setOnClickListener(v -> {
                onItemClick.accept(item.getId());
            });
        }
    }
}

File: app/src/main/java/ru/sicampus/bootcamp2025/ui/list/ListAdapter.kt
package ru.sicampus.bootcamp2025.ui.list

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.fragment.app.FragmentManager
import androidx.paging.PagingDataAdapter
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView
import ru.sicampus.bootcamp2025.R
import ru.sicampus.bootcamp2025.databinding.ItemCenterBinding
import ru.sicampus.bootcamp2025.domain.list.ListEntity
import ru.sicampus.bootcamp2025.domain.one.OneCenter
import ru.sicampus.bootcamp2025.ui.one.OneCenterFragment

class ListAdapter(
    private val fragmentManager: FragmentManager
) : PagingDataAdapter<ListEntity, ListAdapter.ViewHolder>(UserDiff) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            ItemCenterBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            ),
            fragmentManager
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        getItem(position)?.let { holder.bind(it) }
    }

    class ViewHolder(
        private val binding: ItemCenterBinding,
        private val fragmentManager: FragmentManager
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind(item: ListEntity) {
            with(binding) {
                title.text = item.name
                description.text = item.description

                coordinates.text = root.context.getString(
                    R.string.coordinates_format,
                    item.coordinates.latitude,
                    item.coordinates.longitude
                )

                val oneCenter = OneCenter(
                    id = item.id,
                    name = item.name,
                    description = item.description,
                    coordinates = item.coordinates
                )

                binding.actionButton.setOnClickListener {
                    fragmentManager.beginTransaction()
                        .replace(R.id.main, OneCenterFragment.newInstance(oneCenter))
                        .commitAllowingStateLoss()
                }
            }
        }
    }

    object UserDiff : DiffUtil.ItemCallback<ListEntity>() {
        override fun areItemsTheSame(oldItem: ListEntity, newItem: ListEntity): Boolean {
            return oldItem.id == newItem.id
        }

        override fun areContentsTheSame(oldItem: ListEntity, newItem: ListEntity): Boolean {
            return oldItem == newItem
        }
    }
}